examples/basic-http.yaml

```diff
@@ -35,32 +35,23 @@ metadata:
   labels:
     app: foo
 spec:
-  hosts:
-  - hostnames:
-    - "foo.com"
-    rules:
-    - match:
-        path:
-          type: Prefix
-          value: /bar
-      action:
-        forwardTo:
-          - targetRef:
-              name: my-service
-              group: core
-              resource: services
-    - match:
-        headers:
-          type: Exact
-          values:
-            x-forwarded-proto: https
-            magic: foo
-        path:
-          type: Prefix
-          value: /some/thing
-      action:
-        forwardTo:
-          - targetRef:
-              name: my-service
-              group: core
-              resource: services
+  hostnames:
+  - "foo.com"
+  rules:
+  - matches:
```

Comment:
This is out of scope for this PR, but I think we should consider renaming `matches` to `match`. There are other areas in the APIs where a singular field name is being used for a list.